src/createElement.test.ts

```diff
@@ -7,4 +8,24 @@ describe("createElement", () => {
       "Unsupported lower-case element 'test' encountered, please make sure all your components start with an upper-case letter and are functions."
     );
   });
+
+  it("throws an error if a multiple children detected for promise element", () => {
+    expect(() =>
+      // @ts-expect-error - await element test multiple promises as children
+      createElement(Await, {}, Promise.resolve("Test"), Promise.resolve("Test"))
+    ).toThrowError(
+      "<Await> expected to receive a single promise that resolves to any markdown children."
+    );
+  });
+
+  it("creates a await element if a single promise is used", () => {
```

Comment:
Just realized that I've made some grammar mistake here, should be 
> an await element

instead of 

> a await element